BCnEnc.Net/Decoder/BcDecoder.cs

```diff
@@ -35,34 +33,34 @@ public class BcDecoder
 		#region Async Api
 
 		/// <summary>
-		/// Decode raw encoded image asynchronously.
+		/// DecodeInternal raw encoded image asynchronously.
 		/// </summary>
 		/// <param name="inputStream">The stream containing the encoded data.</param>
 		/// <param name="format">The Format the encoded data is in.</param>
 		/// <param name="pixelWidth">The pixelWidth of the image.</param>
 		/// <param name="pixelHeight">The pixelHeight of the image.</param>
 		/// <param name="token">The cancellation token for this asynchronous operation.</param>
 		/// <returns>The awaitable operation to retrieve the decoded Rgba32 image.</returns>
-		public Task<Image<Rgba32>> DecodeRawAsync(Stream inputStream, CompressionFormat format, int pixelWidth, int pixelHeight, CancellationToken token = default)
+		public Task<ColorRgba32[]> DecodeRawAsync(Stream inputStream, CompressionFormat format, int pixelWidth, int pixelHeight, CancellationToken token = default)
```

Comment:
Could also add DecodeToMemory methods for each of these, which would return Memory2D with the pixel data. What do you think?

Reply:
It would make it more symmetrical to the encoding API, so I would agree.

Reply:
I would replace all the ColorRgba32[]'s with ReadOnlyMemory<ColorRgba32> as return values.
This would play along the mentioned code symmetry and the comments by @ptasev below.

Reply:
Returns should be ColorRgba32[] / Memory<ColorRgba32> or Memory2D<ColorRgba32>. I don't think they should be read-only, if the user wants to do something with them after.

Reply:
If you are allocating a brand new array to return, might as well return the array directly. I don't know if you're adding anything useful by wrapping it in a `Memory<T>`. Also, agreed that it should not be read only as you want to allow to user to have complete control over the returned data.